tests/src/test/scala/akka/kafka/scaladsl/RebalanceExtSpec.scala

```diff
@@ -0,0 +1,427 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <https://softwaremill.com>
+ * Copyright (C) 2016 - 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.kafka.scaladsl
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import akka.actor.{Actor, ActorLogging, DeadLetter, Props}
+import akka.kafka.ConsumerMessage.{CommittableMessage, CommittableOffset}
+import akka.kafka._
+import akka.kafka.testkit.scaladsl.TestcontainersKafkaLike
+import akka.stream._
+import akka.stream.scaladsl.{Flow, Keep, Sink, Source}
+import akka.stream.testkit.scaladsl.StreamTestKit.assertAllStagesStopped
+import akka.{Done, NotUsed}
+import org.apache.kafka.clients.consumer.ConsumerConfig
+import org.apache.kafka.common.TopicPartition
+import org.scalatest._
+import org.slf4j.Logger
+
+import scala.collection.mutable.{Set => MSet}
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Future, Promise}
+import scala.util.Try
+
+class RebalanceExtSpec extends SpecBase with TestcontainersKafkaLike with Inside {
+
+  implicit override val patienceConfig: PatienceConfig = PatienceConfig(30.seconds, 500.millis)
+  implicit val maxAwait: Duration = 10.minute
```

Comment:
Could we use `remainingOrDefault` here to make timeouts configurable?

Reply:
done